src/config.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright The ocicrypt Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+
+/// CryptoConfig holding the identifiers and public/private key file data
+/// that will be able to encrypt/decrypt the symmetric key
+#[derive(Debug)]
+pub struct CryptoConfig {
+    /// encrypt map holding 'gpg-recipients', 'gpg-pubkeyringfile', 'pubkeys', 'enc-x509s'
+    /// decrypt map holding 'privkeys', 'dec-x509s', 'gpg-privatekeys'
+    pub param: HashMap<String, Vec<Vec<u8>>>,
+}
+
+impl CryptoConfig {
+    /// Return a new CryptoConfig instance.
+    pub fn new() -> Self {
+        CryptoConfig {
+            param: HashMap::new(),
```

Comment:
Crypto config should have 2 sub fields for `EncryptConfig` and `DecryptConfig`, this is to help provide context into which private/public material should be, and to make sure that only the right content is passed onto the right components.

Reply:
Yes, for the Crypto config, I also not sure the best way to define them. We may have three options: 1. like current go version, we define three struct: `EncryptConfig`, `DecryptConfig`, `CryptioConfig`, but it seems a bit duplicate, the DecryptConfig in CryptoConfig will always be a reference in EncryptConfig. 2. Define one struct: `CryptoConfig`,with two hashmaps with different names like encrypt_param and decrypt_param to seperate them 3. current definication with one struct and one hashmap, we seperate different value through different hash keys which is more simple. Any suggestions for these three options?  BTW，the user will use the create_encrypt/decrypt_config API to get their CryptoConfig instances,  the private/public material will be seperated by different CryptoConfig instances.

Reply:
I do think it is a valid point about streamlining this struct.. However, since construction of the struct is hidden behind the interface and it is a public interface, I am leaning towards having this be consistent across the implementations and if we want to redefine it, we should do it for the golang version as well.

Reply:
Yes, we should keep the compatiable between ocicrypt golang and rust version and try to make the API consistent. But during search the rust crypto llibraries, I found many crypto rust implementations don't have the same API with the golang version. Also rust has ownership feature to make memory safe, the pointer in current  `CryptoConfig` struct need under the same scope with the real data, it will restrict current usage to return `CryptoConfig` only in create config API. We will ensure the container image encrypted by golang version ocicrypt can be decrypted by rust version ocicrypt and vice versa. I not sure whether we can adjust this config rust API first, after we finished the rust work, we can submit patches for golang version?

Reply:
> I think the main thing that we need discussion here is on the convention of string keys within the map to be consistent with the golang implementation - since there is an implicit contract here across languages, especially with keyprovider with the ability to do this across gRPC/exec.

I believe it would make sense to at least have 2 maps in `CryptoConfig` (`encrypt_params` and `decrypt_params`), if we don't want to have 2 separate structures.

Reply:
If we want to have something equivalent, API-wise, to the golang implementation, I think it should look like this:

```Rust
struct DecryptConfig {
     param: HashMap<String, Vec<Vec<u8>>>,
}
struct EncryptConfig {
     param: HashMap<String, Vec<Vec<u8>>>,
     decrypt_config: DecryptConfig,
}

struct CryptoConfig {
     encrypt_config: Option<EncryptConfig>,
     decrypt_config: Option<DecryptConfig>,
}
```

Reply:
Thanks @lumjjb @sameo , if we need the string keys within the map to be consistent with the golang implementation, we need two maps since we have same string x509s in both map. If we want API consistent, I also agree with @sameo 's second approch with sperate struct. Since `DecryptConfig` in `EncryptConfig` is optional, we can also make it as a `Option`:
```rust
#[derive(Debug, Default, Clone)]
pub struct EncryptConfig {
     /// map holding 'gpg-recipients', 'gpg-pubkeyringfile', 'pubkeys', 'enc-x509s'
     pub param: HashMap<String, Vec<Vec<u8>>>,

     /// Allow for adding wrapped keys to an encrypted layer.
     pub decrypt_config: Option<DecryptConfig>,
}
```  
With this approch, it is flexible and no duplicate.

Reply:
I just update a version with this definition, please review.

Reply:
This looks pretty good to me. @lumjjb WDYT?

Reply:
Looks great! :+1: thanks!

Reply:
I'm happy to merge that PR, I think it's a good first step. The immediate next one would be to add a simple gh action based CI. @lumjjb Are you ok if we proceed that way?